### ISSUE TITLE
Save extra data to marker (or layer)

### ISSUE THREAD

**Issue by steosoft** [mstahv/g-leaflet#58]
It should be possible to add custom extra data to marker.
In original library we can do it just by passing key-value while creating marker object.
Is it possible to add this function to library?

**Comment by mstahv** [comment]
You can extend Marker class and create a custom version. Then you can add either well typed api for properties you know or even expose a map like API if you really need that.

**Comment by octavm** [comment]
Have you thought about using the `setData()` method of `LMarker`, inherited from `AbstractComponent`?

**Comment by mstahv** [comment]
I guess @steosoft is using this in GWT application? The setData suggested by @octavm only in the Vaadin integration side. Even there I'd suggest to create the extensions for better typed and documented API.

**Comment by octavm** [comment]
@mstahv  you're right. I wasn't paying attention, just now I noticed this is for g-leaflet.

**Comment by steosoft** [comment]
Yes, I'm using it in GWT application. I've changed my code so I don't need passing extra data anymore but good to know how to do that in the future. Thank you.
